src/store.rs

```diff
@@ -31,173 +28,24 @@ impl StoreParams for DefaultStoreParams {
-        self.ops.len()
-    }
-
-    /// Increases the pin count of a block.
-    pub fn pin<I: Into<Cow<'a, Cid>>>(&mut self, cid: I) {
-        self.ops.push_back(Op::Pin(cid.into()));
-    }
-
-    /// Decreases the pin count of a block.
-    pub fn unpin<I: Into<Cow<'a, Cid>>>(&mut self, cid: I) {
-        self.ops.push_back(Op::Unpin(cid.into()));
-    }
-
-    /// Update a block.
-    ///
-    /// Pins the new block and unpins the old one.
-    pub fn update<I: Into<Cow<'a, Cid>>, J: Into<Cow<'a, Cid>>>(&mut self, old: Option<I>, new: J) {
-        self.pin(new);
-        if let Some(old) = old {
-            self.unpin(old);
-        }
-    }
-
-    /// Imports a block.
-    ///
-    /// This will add the block to the front of the transaction.
-    ///
-    /// When importing blocks from a network, they'll be fetched in
-    /// reverse order. This ensures that when inserting the block
-    /// all it's references have been inserted.
-    pub fn import(&mut self, block: Block<S>) -> Result<()>
-    where
-        Ipld: Decode<S::Codecs>,
-    {
-        let refs = block.references()?;
-        self.ops.push_front(Op::Insert(block, refs));
-        Ok(())
-    }
-
-    /// Inserts a block.
-    ///
-    /// This will add the block to the end of the transaction.
-    pub fn insert(&mut self, block: Block<S>) -> Result<()>
-    where
-        Ipld: Decode<S::Codecs>,
-    {
-        let refs = block.references()?;
-        self.ops.push_back(Op::Insert(block, refs));
-        Ok(())
-    }
-
-    /// Creates a block.
-    ///
-    /// This will add the block to the end of the transaction.
-    ///
-    /// When constructing blocks, the all references need to be
-    /// created before the referrer.
-    pub fn create<CE: Codec, T: Encode<CE> + ?Sized>(
-        &mut self,
-        codec: CE,
-        hash: u64,
-        value: &T,
-    ) -> Result<Cid>
-    where
-        CE: Into<S::Codecs>,
-        Ipld: Decode<S::Codecs>,
-    {
-        let block = Block::<S>::encode(codec, hash, value)?;
-        let cid = block.cid().clone();
-        self.insert(block)?;
-        Ok(cid)
-    }
-}
-
-impl<'a, S: StoreParams> IntoIterator for &'a Transaction<'a, S> {
-    type Item = &'a Op<'a, S>;
-    type IntoIter = std::collections::vec_deque::Iter<'a, Op<'a, S>>;
-
-    fn into_iter(self) -> Self::IntoIter {
-        self.ops.iter()
-    }
-}
-
-impl<'a, S: StoreParams> IntoIterator for Transaction<'a, S> {
-    type Item = Op<'a, S>;
-    type IntoIter = std::collections::vec_deque::IntoIter<Self::Item>;
-
-    fn into_iter(self) -> Self::IntoIter {
-        self.ops.into_iter()
-    }
-}
-
-/// Implementable by ipld stores.
+/// Implementable by ipld stores. An ipld store behaves like a cache. It will keep blocks
+/// until the cache is full after which it evicts blocks based on an eviction policy. If
+/// a block is aliases, it and it's recursive references will not be evicted or counted towards
+/// the cache size.
```

Comment:
I would add here that an alias is a named pin, since the concept of pinning is familiar to people coming from go-ipfs.